runtime/server_http_response.go

```diff
@@ -81,23 +75,8 @@ func (res *ServerHTTPResponse) finish() {
 		/* coverage ignore next line */
 		return
 	}
-
 	res.finished = true
 	res.finishTime = time.Now()
-
-	counter := res.metrics.statusCodes[res.StatusCode]
-	if counter == nil {
-		res.Request.Logger.Error(
-			"Could not emit statusCode metric",
-			zap.Int("UnexpectedStatusCode", res.StatusCode),
-		)
-	} else {
-		counter.Inc(1)
-	}
-
-	res.metrics.requestLatency.Record(
```

Comment:
Let's keep the metric emission here in the `finish()` method.

Reply:
Ok.  As we discussed offline.